### PR TITLE
Fix Cmake files to select correct engine, upgrade icons etc

### PR DESCRIPTION
**Pull request by lmbr-pip**
* Fix engine finder cmake to match MPS so correct engine is selected (was seeing wrong engine constantly being used locally)
* Add new O3DE logo icons to project
* Minor compiler warning fixes
* Tweaks to README

## Testing
- [x]  Build netsoak test on windows and ran with/without null rendering
- [x]  Build netsoak test on Linux and run with/without null rendering

Signed-off-by: Potter <rppotter@amazon.com>

### CODE REVIEW

File: Gem/Code/Source/NetSoakTestSystemComponent.cpp [hunk 1]
```diff
@@ -93,8 +93,8 @@ namespace NetSoakTest
     static const AZStd::string_view s_networkInterfaceName("NetSoakNetworkInterface");
     static const AZStd::string_view s_loopbackInterfaceName("NetSoakLoopbackInterface");
 
-    AZ_CVAR(AZ::TimeMs, soak_latencyms, AZ::TimeMs(0), nullptr, AZ::ConsoleFunctorFlags::DontReplicate, "Simulated connection quality latency");
-    AZ_CVAR(AZ::TimeMs, soak_variancems, AZ::TimeMs(0), nullptr, AZ::ConsoleFunctorFlags::DontReplicate, "Simulated connection quality variance");
+    AZ_CVAR(AZ::TimeMs, soak_latencyms, static_cast<AZ::TimeMs>(0), nullptr, AZ::ConsoleFunctorFlags::DontReplicate, "Simulated connection quality latency");
```
Comment (AMZN-Gene) [inline Gem/Code/Source/NetSoakTestSystemComponent.cpp:96]: AZ::Time::ZeroTimeMs

File: Gem/Code/Source/NetSoakTestSystemComponent.cpp [hunk 2]
```diff
@@ -93,8 +93,8 @@ namespace NetSoakTest
...
+    AZ_CVAR(AZ::TimeMs, soak_variancems, static_cast<AZ::TimeMs>(0), nullptr, AZ::ConsoleFunctorFlags::DontReplicate, "Simulated connection quality variance");
```
Comment (AMZN-Gene) [inline Gem/Code/Source/NetSoakTestSystemComponent.cpp:97]: AZ::Time::ZeroTimeMs

**Review by AMZN-Gene — APPROVED** [review]

**Review by AMZN-puvvadar — APPROVED** [review]

File: Gem/Code/Source/NetSoakTestSystemComponent.cpp (hunk 1 above)
Comment (lmbr-pip) [reply Gem/Code/Source/NetSoakTestSystemComponent.cpp]: Added

File: Gem/Code/Source/NetSoakTestSystemComponent.cpp [hunk 3]
```diff
@@ -199,7 +199,7 @@ namespace NetSoakTest
                 unreliable.SetSmallDatum(2);
                 connection.SendUnreliablePacket(unreliable);
             }
-            
+
```
Comment (AMZN-puvvadar) [inline Gem/Code/Source/NetSoakTestSystemComponent.cpp:202]: NP: Looks like a whitespace snuck in

File: README.md [hunk 4]
```diff
@@ -118,6 +118,7 @@ Note: All O3DE projects generate a GameLauncher and a ServerLauncher. NetSoakTes
 | soak_port | The port that this soak test will bind to for game traffic | 33450 |
 | soak_protocol | Soak test protocol (TCP or UDP) | udp | 
 | soak_mode | The operating mode for the soak test, options are loopback, client or host. `Loopback` has two connection within the application feed traffic to each other in a loop. `Client` expects to connect to a server hosted at soak_serveraddr. `Host` hosts a server for clients to connect to | Loopback | 
+| DumpSoakStats | Dump snapshot of soak test networking stats to console | N/A|
```
Comment (AMZN-puvvadar) [inline README.md:121]: NP: Missing a trailing space after N/A. That might not produce any difference in the README

File: Gem/Code/Source/NetSoakTestSystemComponent.cpp (hunk 3 above)
Comment (lmbr-pip) [reply Gem/Code/Source/NetSoakTestSystemComponent.cpp:202]: It was removed ;)

File: README.md (hunk 4 above)
Comment (lmbr-pip) [reply README.md:121]: Happy to fix that in next change, if ok as am adding option.

**Pull request merged**